administrator/components/com_associations/models/associations.php

```diff
@@ -0,0 +1,234 @@
+
+			if ($position = strpos($componentModel, 'getAssociations'))
+			{
+				// Searching for , '#__table' , after getAssociations(
+				$start = strpos($componentModel, ',', $position) + 2;
+				$end = strpos($componentModel, ',', $start) - 1;
+				
+				if ($componentSplit[0] == 'com_menus')
+				{
+					$table = '#__menu';
+				}
+				else 
+				{
+					$table = str_replace("'", "", substr($componentModel, $start, $end - $start));
+				}
+			}	
+		}
+		// If it's a category
+		elseif (strpos($component, '|'))
+		{
+			$componentSplit = explode('|', $component);
+			$extension      = $componentSplit[1];
+			$table          = '#__categories';
+		}
+
+		$columns  = $db->getTableColumns($table);
+		$title    = isset($columns['title']) ? 'a.title' : 'a.name';
+		$ordering = isset($columns['lft']) ? 'a.lft' : 'a.ordering';
+
+		$query->select('a.id, ' . $title . ' AS title, a.language, ' . $ordering . ' AS ordering');
+		$query->from($db->quoteName($table, 'a'));
+
+		// Join over the language
+		$query->select('l.title AS language_title, l.image AS language_image')
+			->join('LEFT', $db->quoteName('#__languages') . ' AS l ON l.lang_code = a.language');
+
+		// Join over the associations.
+		$query->select('COUNT(' . $db->quoteName('asso2.id') . ') > 1 as ' . $db->quoteName('association'))
+			->join(
+				'LEFT',
+				$db->quoteName('#__associations', 'asso') . ' ON ' . $db->quoteName('asso.id') . ' = ' . $db->quoteName('a.id')
+				. ' AND ' . $db->quoteName('asso.context') . ' = ' . $db->quote($componentSplit[0] . '.item')
+			)
+			->join(
+				'LEFT',
+				$db->quoteName('#__associations', 'asso2') . ' ON ' . $db->quoteName('asso2.key') . ' = ' . $db->quoteName('asso.key')
+			)
+			->group(
+				$db->quoteName(
+					array(
+						'a.id',
+						'title',
+						'a.language'
+					)
+				)
+			);
+
+		if ($table == '#__menu')
+		{
+			// Exclude the root category.
+			$query->where('a.id > 1')
+				->where('a.client_id = 0');
+		}
+
+		if ($table == '#__categories')
+		{
+			$query->where('a.extension = ' . $db->quote($extension));
+		}
+
+		// Filter on the language.
+		if ($language = $this->getState('associationlanguage'))
+		{
+			$query->where($db->quoteName('a.language') . ' = ' . $db->quote($language));
+		}
+
+		// Filter by search in name.
+		$search = $this->getState('filter.search');
+		if (!empty($search))
+		{
+			if (stripos($search, 'id:') === 0)
+			{
+				$query->where('a.id = ' . (int) substr($search, 3));
+			}
+			else
+			{
+				$search = $db->quote('%' . str_replace(' ', '%', $db->escape(trim($search), true) . '%'));
+				$query->where(
+					'(' . $db->quoteName('title') . ' LIKE ' . $search . ')'
+				);
+			}
+		}
+
+		// Add the list ordering clause.
+		$orderCol = $this->state->get('list.ordering', 'title');
+		$orderDirn = $this->state->get('list.direction', 'asc');
+
+		if ($orderCol == 'ordering')
+		{
+			$orderCol = 'title ' . $orderDirn . ', ordering';
+		}
```

Comment:
what's th epoitn in ordering by order, if first order by title? or am i seeing this wrong?